### PR TITLE
Generate reports per run, per project and per lane

### LINKED ISSUE #3: Generate reports per run, per project and per lane
### Description of feature

MultiQC should be run several times to generates report on the following subsets of the data:
* One report per flowcell
* One report per lane
* One report per project

A similar feature is implemented in the seqreport pipeline: https://github.com/Molmed/seqreports/blob/main/main.nf

### PR DESCRIPTION
**Pull request by Aratz**
This PR introduces MultiQC report generation by lane, by rundir and by sample group.

Closes #3 

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, also make a PR on the nf-core/seqinspector _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by Aratz** [comment]
Not quite, the tests need fixing and I need to go through the rest of the PR checklist (but the code is there).

File: subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf
[the remark this answers is not in the document]
Comment (MatthiasZepper) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:1]: I have at least realized that they are required 😄 ...only the magic wizardry that happens there is still obscure❓ 

```
Caused by:
  Not a valid path value type: java.util.ArrayList ([/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample23_S3_L001_R1_001.fastq.gz])
```

**Comment by MatthiasZepper** [comment]
If I `.dump()` the `ch_samplesheet`, neither the _id:_ nor the path to the FastQ file are quoted. Should they be strings, or is that intentional? 
```

[DUMP: samplesheet] [['sample':'Sample1', 'lane':1, 'group':'S1', 'rundir':null, 'id':Sample1_S1_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample1_S1_L001_R1_001.fastq.gz]]
[DUMP: samplesheet] [['sample':'SampleA', 'lane':1, 'group':'S2', 'rundir':null, 'id':SampleA_S2_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/SampleA_S2_L001_R1_001.fastq.gz]]
[DUMP: samplesheet] [['sample':'Sample23', 'lane':1, 'group':'S3', 'rundir':null, 'id':Sample23_S3_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample23_S3_L001_R1_001.fastq.gz]]
[DUMP: samplesheet] [['sample':'sampletest', 'lane':1, 'group':'S4', 'rundir':null, 'id':sampletest_S4_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/sampletest_S4_L001_R1_001.fastq.gz]]
[DUMP: samplesheet] [['sample':'Undetermined', 'lane':1, 'group':null, 'rundir':null, 'id':Undetermined_ungrouped_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Undetermined_S0_L001_R1_001.fastq.gz]]
```

**Comment by mahesh-panchal** [comment]
> If I `.dump()` the `ch_samplesheet`, neither the _id:_ nor the path to the FastQ file are quoted. Should they be strings, or is that intentional?
> 
> ```
> 
> [DUMP: samplesheet] [['sample':'Sample1', 'lane':1, 'group':'S1', 'rundir':null, 'id':Sample1_S1_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample1_S1_L001_R1_001.fastq.gz]]
> [DUMP: samplesheet] [['sample':'SampleA', 'lane':1, 'group':'S2', 'rundir':null, 'id':SampleA_S2_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/SampleA_S2_L001_R1_001.fastq.gz]]
> [DUMP: samplesheet] [['sample':'Sample23', 'lane':1, 'group':'S3', 'rundir':null, 'id':Sample23_S3_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample23_S3_L001_R1_001.fastq.gz]]
> [DUMP: samplesheet] [['sample':'sampletest', 'lane':1, 'group':'S4', 'rundir':null, 'id':sampletest_S4_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/sampletest_S4_L001_R1_001.fastq.gz]]
> [DUMP: samplesheet] [['sample':'Undetermined', 'lane':1, 'group':null, 'rundir':null, 'id':Undetermined_ungrouped_1, 'single_end':true], [/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Undetermined_S0_L001_R1_001.fastq.gz]]
> ```

I'm not certain but I think `id` is not quoted because it's a `GString`, and I think Path objects aren't quoted either as they're not strings

File: subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf [hunk 1]
```diff
@@ -84,7 +84,7 @@ workflow PIPELINE_INITIALISATION {
         .fromSamplesheet("input") // Validates samplesheet against $projectDir/assets/schema_input.json. Path to validation schema is defined by $projectDir/nextflow_schema.json
         .map {
             meta, fastq_1, fastq_2 ->
-                def id_string = "${meta.sample}_${meta.project ?: "ungrouped"}_${meta.lane}"
+                def id_string = "${meta.sample}_${meta.group ?: "ungrouped"}_${meta.lane}"
```
[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: It's been removed from required.

[the remark this answers is not in the document]
Comment (MatthiasZepper) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: 😳 Have I then been reviewing the wrong/outdated version of this PR all the time? Because I ran `gh pr 13 checkout` and it is still in there for me locally ?!?

**Comment by MatthiasZepper** [comment]
> I'm not certain but I think `id` is not quoted because it's a `GString`, and I think Path objects aren't quoted either as they're not strings

I see. I mainly noticed it, because I wanted to create a minimally reproducible example to try out that other pattern that I had in mind. Thus, I copy-pasted the dump into a single `main.nf` script and used a `Channel.of()` factory, and it would not work (and produced an extremely unhelpful error message) unless I quoted everything.

File: subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (kedhammar) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: I removed it so we would be able to run on sequencing platforms without lanes, e.g. ONT.

**Comment by kedhammar** [comment]
nf-test is not working for me since the last 3 commits, it seems `params.pipelines_testdata_base_path` evaluates to `null`

File: subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (kedhammar) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: > Doesn't lane need a default value too if it's not required?

The way I understand it from the code in `workflows/seqinspector.nf`, the `filter` operation means it simply doesn't perform lane-wise multiQC if the field is empty.

```
    multiqc_extra_files_per_group = ch_multiqc_files
        .filter { meta, sample -> meta.group }
        .map { meta, sample -> meta.group }
        .unique()
        .map { group -> [group:group] }
        .combine(ch_multiqc_extra_files)
```

File: workflows/seqinspector.nf
[the remark this answers is not in the document]
Comment (MatthiasZepper) [reply workflows/seqinspector.nf:1]: Because those `.filter()` and  `.multiMap()` chains seemed unnecessarily complex, I experimented with an own solution (including the `.cross()` operator, which I then dropped again), but ultimately the differences aren't large. Like you, I ended up moving the grouping variables out of the meta map and grouped over them. 

Most of what makes your solution seemingly more complicated is the juggling with MultiQC config files, which I have not included in my minimal example, so it is an unfair competition. On the plus side, I have only now understood your approach, I believe :-) 

```
#!/usr/bin/env nextflow

workflow {
ch_samplesheet = Channel.of(
[['sample':'SampleA', 'group':'S1', 'lane':'1' ], ['/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample1_S1_L001_R1_001.fastq.gz']],
[['sample':'SampleB', 'group':'S1', 'lane':'2'], ['/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/SampleA_S2_L001_R1_001.fastq.gz']],
[['sample':'SampleC', 'group':'S2', 'lane':'1'], ['/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Sample23_S3_L001_R1_001.fastq.gz']],
[['sample':'SampleD', 'group':'S2', 'lane':'1'], ['/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/sampletest_S4_L001_R1_001.fastq.gz']],
[['sample':'Undetermined', 'group':null, 'lane':'1'], ['/nf-core/test-datasets/raw/seqinspector/testdata/NovaSeq6000/200624_A00834_0183_BHMTFYDRXX/Undetermined_S0_L001_R1_001.fastq.gz']]
)


// ------------------------------------------------------
// Apply the various QC Tools to that same input channel
// ------------------------------------------------------

ch_qc_outputs = some_qc_tool(ch_samplesheet)
// Mix everything together: No problem as long as the meta remains intact
// I tried join here, but it does not tolerate null values apparently.
ch_qc_outputs = ch_qc_outputs.mix(some_other_qc_tool(ch_samplesheet))


// ----------------------------------------------------------------------------------------------------------------------------------
// At the very end, we move the grouping variables to the front and groupTuples based on combination of all possible grouping levels
// ----------------------------------------------------------------------------------------------------------------------------------

// we also simplify the meta to the sample name, the only thing still needed.
ch_qc_outputs_final = ch_qc_outputs.map{ meta, sample -> [ "${meta.group}", "${meta.lane}", ["${meta.sample}", sample]]}.groupTuple(by: [0,1])

// -----------------------------------------------------------------------------------------------
// Group again to the desired level (e.g. lanes)
// -----------------------------------------------------------------------------------------------

ch_qc_outputs_lane_subsets = ch_qc_outputs_final.groupTuple(by: [1])

ch_qc_outputs_group_subsets = ch_qc_outputs_final.groupTuple(by: [0])

}

process some_qc_tool {
    input: 
        tuple val(meta), path(fastq)
    output:
         tuple val(meta), path("*.log"), emit: qc
    script:
    """
    echo "QC of $fastq" > qc_stats.log
    """
}

process some_other_qc_tool {
    input: 
        tuple val(meta), path(fastq)
    output:
        tuple val(meta), path("*.log"), emit: qc
    script:
    """
    echo "QC of $fastq" > qc_stats.log
    """
}
```

File: docs/usage.md [hunk 2]
```diff
@@ -16,39 +16,27 @@ You will need to create a samplesheet with information about the samples you wou
...
-sample,fastq_1,fastq_2
-CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
-CONTROL_REP2,AEG588A2_S2_L002_R1_001.fastq.gz,AEG588A2_S2_L002_R2_001.fastq.gz
-CONTROL_REP3,AEG588A3_S3_L002_R1_001.fastq.gz,AEG588A3_S3_L002_R2_001.fastq.gz
-TREATMENT_REP1,AEG588A4_S4_L003_R1_001.fastq.gz,
-TREATMENT_REP2,AEG588A5_S5_L003_R1_001.fastq.gz,
-TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,
-TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,
+sample,lane,group,fastq_1,fastq_2,rundir
+CONTROL_REP1,1,,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz,200624_A00834_0183_BHMTFYDRXX
```
[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply docs/usage.md]: Very nice. This definitely helps.

File: subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: Sure, but what about the other paths, e.g. channel where `meta.group` has a setting, but `meta.lane` has nothing (and the filter is on `meta.group`)?
The name will include `null` in it.

[the remark this answers is not in the document]
Comment (kedhammar) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: Could you point out where this would be an issue?

I'll note that I don't mind re-working this code into something more explicit, I simply lack the know-how as of now 😆

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: I guess the question is, is `id` used for anything important: For example with the Promethion test, the csv looks like:
```
sample,lane,group,fastq_1,fastq_2,rundir
hg001,,r10p41_e8p2_human_runs_jkw,https://github.com/nf-core/test-datasets/raw/seqinspector/testdata/PromethION/20230505_1857_1B_PAO99309_94e07fab/fastq_pass/PAO99309_pass__94e07fab_c3641428_1.fastq.gz,,
```
and then the id string should be: `hg001_r10p41_e8p2_human_runs_jkw_null`. Does it matter that this is the case?
When you're grouping the files by group `r10p41_e8p2_human_runs_jkw` I guess lane information is not needed at all downstream of this?

[the remark this answers is not in the document]
Comment (kedhammar) [reply subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf:87]: Thanks for clarifying! From what I remember of the initial meeting, the `id` was intended as simply a concatenation of fields to ensure uniqueness within the pipeline run. This is still ensured even if some of the fields are null, right?

Intuitively I think having a consistent way to generate the `id` that sometimes contains null is preferable to setting up different conventions for generating it across different sequencing platforms.

File: workflows/seqinspector.nf [hunk 3]
```diff
@@ -50,45 +57,163 @@ workflow SEQINSPECTOR {
...
-    ch_multiqc_files = ch_multiqc_files.mix(
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(ch_collated_versions)
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(
         ch_methods_description.collectFile(
             name: 'methods_description_mqc.yaml',
             sort: true
         )
     )
 
     MULTIQC (
```
[the remark this answers is not in the document]
Comment (kedhammar) [reply workflows/seqinspector.nf:87]: https://github.com/nf-core/seqinspector/pull/13#discussion_r1601867747

File: workflows/seqinspector.nf [hunk 4]
```diff
@@ -4,12 +4,17 @@
...
 
-include { FASTQC                 } from '../modules/nf-core/fastqc/main'
-include { MULTIQC                } from '../modules/nf-core/multiqc/main'
-include { paramsSummaryMap       } from 'plugin/nf-validation'
-include { paramsSummaryMultiqc   } from '../subworkflows/nf-core/utils_nfcore_pipeline'
-include { softwareVersionsToYAML } from '../subworkflows/nf-core/utils_nfcore_pipeline'
-include { methodsDescriptionText } from '../subworkflows/local/utils_nfcore_seqinspector_pipeline'
+include { FASTQC                        } from '../modules/nf-core/fastqc/main'
+
+include { MULTIQC                       } from '../modules/nf-core/multiqc/main'
```
[the remark this answers is not in the document]
Comment (kedhammar) [reply workflows/seqinspector.nf]: [84c9b3d](https://github.com/nf-core/seqinspector/pull/13/commits/84c9b3d95c693b722a3e61752ba498b05fc08e57)

File: workflows/seqinspector.nf [hunk 5]
```diff
@@ -50,45 +57,163 @@ workflow SEQINSPECTOR {
...
+        Channel.empty().toList(),
+        ch_multiqc_logo.toList()
+    )
+
+    // Generate reports by lane
+    multiqc_extra_files_per_lane = ch_multiqc_files
+        .filter { meta, sample -> meta.lane }
+        .map { meta, sample -> meta.lane }
+        .unique()
+        .map { lane -> [lane:lane] }
```
[the remark this answers is not in the document]
Comment (kedhammar) [reply workflows/seqinspector.nf]: [84c9b3d](https://github.com/nf-core/seqinspector/pull/13/commits/84c9b3d95c693b722a3e61752ba498b05fc08e57)

**Comment by kedhammar** [comment]
I've tried to clean up the discussion thread and have pushed some additional commits to address simple issues. Requesting re-reviews.

**Review by kedhammar — APPROVED** [review]

File: workflows/seqinspector.nf [hunk 6]
```diff
@@ -50,45 +57,163 @@ workflow SEQINSPECTOR {
...
+            lane, meta, samples -> [
+                "${lane}_multiqc_extra_config.yml",
+                """
+                    |output_fn_name: \"${lane}_multiqc_report.html\"
+                    |data_dir_name:  \"${lane}_multiqc_data\"
+                    |plots_dir_name: \"${lane}_multiqc_plots\"
+                """.stripMargin()
+            ]
+        }
+        .map { file -> [ (file =~ /(\[LANE:.+\])/)[0][1], file ] }
```
[the remark this answers is not in the document]
Comment (Aratz) [reply workflows/seqinspector.nf:120]: I'm working on this now but it's taking some time, fighting against nextflow error messages again...

File: workflows/seqinspector.nf [hunk 7]
```diff
@@ -50,45 +57,162 @@ workflow SEQINSPECTOR {
...
+        .map { group -> [group:group] }
+        .combine(ch_multiqc_extra_files)
+
+    group_mqc_files = ch_multiqc_files
+        .filter { meta, sample -> meta.group }
+        .mix(multiqc_extra_files_per_group)
+        .map { meta, sample -> [ "[GROUP:${meta.group}]", meta, sample ] }
+        .groupTuple()
+        .tap { mqc_by_group }
+        .collectFile{
```
Comment (MatthiasZepper) [inline workflows/seqinspector.nf:148]: Is it really required to construct a custom MultiQC config just to set the output file paths? I somehow think that it should be possible to handle that in the `publishDir` of the `module.config`? Or am I missing something?

**Review by MatthiasZepper — APPROVED** [review]
I still have a few open questions, but I also think that could be changed in a subsequent refactor if desired. Thus, I suggest merging and tackle that later?

File: conf/modules.config [hunk 8]
```diff
@@ -27,7 +27,37 @@ process {
         publishDir = [
             path: { "${params.outdir}/multiqc" },
             mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+            saveAs: {
```
[the remark this answers is not in the document]
Comment (Aratz) [reply conf/modules.config:30]: You're right, have some more changes to this file coming up :+1:

File: docs/usage.md [hunk 9]
```diff
@@ -10,47 +10,45 @@
...
 ```csv title="samplesheet.csv"
-sample,fastq_1,fastq_2
-CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
-CONTROL_REP2,AEG588A2_S2_L002_R1_001.fastq.gz,AEG588A2_S2_L002_R2_001.fastq.gz
-CONTROL_REP3,AEG588A3_S3_L002_R1_001.fastq.gz,AEG588A3_S3_L002_R2_001.fastq.gz
-TREATMENT_REP1,AEG588A4_S4_L003_R1_001.fastq.gz,
-TREATMENT_REP2,AEG588A5_S5_L003_R1_001.fastq.gz,
-TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,
-TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,
+sample  lane  group   fastq_1                                       fastq_2 rundir
```
[the remark this answers is not in the document]
Comment (Aratz) [reply docs/usage.md:34]: I don't have a strong opinion on this, and I feel this deserves to be discussed in a new issue/pr. This pr was not meant to change the input format. This commit to `usage.md` just fixes the documentation so that it's up to date with what the format actually is.

File: docs/usage.md [hunk 10]
```diff
@@ -10,47 +10,45 @@
...
-The `sample` identifiers have to be the same when you have re-sequenced the same sample more than once e.g. to increase sequencing depth. The pipeline will concatenate the raw reads before performing any downstream analysis. Below is an example for the same sample sequenced across 3 lanes:
+The following simple run dir structure...
 
-```csv title="samplesheet.csv"
-sample,fastq_1,fastq_2
-CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
-CONTROL_REP1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz
-CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz
+```
+run_dir
```
[the remark this answers is not in the document]
Comment (Aratz) [reply docs/usage.md:24]: @kedhammar ?

File: workflows/seqinspector.nf [hunk 11]
```diff
@@ -50,45 +57,162 @@ workflow SEQINSPECTOR {
...
+
+    MULTIQC_PER_LANE(
+        lane_mqc_files.samples_per_lane,
+        ch_multiqc_config.toList(),
+        lane_mqc_files.config,
+        ch_multiqc_logo.toList()
+    )
+
+    // Generate reports by group
+    multiqc_extra_files_per_group = ch_multiqc_files
```
[the remark this answers is not in the document]
Comment (Aratz) [reply workflows/seqinspector.nf:135]: Correct me if I'm wrong, but when a process or workflow (in this case MULTIQC) is run on two or more queue channels, it will try to zip them and run on each pair of values.

If you don't duplicate the multiqc extra file to follow the same grouping as in the first channel, nextflow will run the lane 1 files against the first multiqc extra file, then the lane 2 files against the second multiqc extra file and so on.

This was very hard to get right, I'm all ears if you see a better way to do it :)

File: workflows/seqinspector.nf (hunk 7 above)
Comment (Aratz) [reply workflows/seqinspector.nf:148]: The issue here is that if you run the MultiQC module once for each lane without specifying different configs each time, it will create files with the same name regardless the lane number. Since the filename is all you have to play with when setting `publishDir`, it becomes very hard to sort them out into different folders.

File: workflows/seqinspector.nf (hunk 6 above)
[the remark this answers is not in the document]
Comment (Aratz) [reply workflows/seqinspector.nf:120]: I've continued a bit with this but didn't have time to get very far. I still think it would be nice to refactor this to avoid duplicating the logic three times. I think I'll make a new PR for this once it is ready, as @MatthiasZepper suggested.

**Review by mahesh-panchal — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -39,26 +39,19 @@
...
-     Explain what rows and columns represent. For instance (please edit as appropriate):
-
 First, prepare a samplesheet with your input data that looks as follows:
 
 `samplesheet.csv`:
 
 ```csv
-sample,fastq_1,fastq_2
-CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
+sample,lane,group,fastq_1,fastq_2,rundir
```
Comment (maxulysse) [inline README.md:47]: what about an extra "individual" field for when you have multiple samples from the same patient (thinking cancer sample sarek style)?

Comment (maxulysse) [reply README.md:47]: or is this what you mean by group?

File: docs/usage.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (kedhammar) [reply docs/usage.md:24]: Since @mahesh-panchal requested we visualize the directory structure, I thought it would be easier to connect the dots to the example samplesheet if all the file names in the dir also contained the information shown in the samplesheet.

I don't necessarily get the impression we are suggesting the files need to follow a particular naming convention by showing an example that is as informative as possible, but I don't feel too strongly about it.

**Comment by Aratz** [comment]
I'll merge this now, thank you all for your reviews and comments. There are still some open discussions that I think are worth addressing but are not critical to this feature, we can keep discussing them here and in subsequent PRs.

File: workflows/seqinspector.nf (hunk 11 above)
[the remark this answers is not in the document]
Comment (Aratz) [reply workflows/seqinspector.nf:135]: I realized I confused myself with my explanation (which for sure is sign that this should be simplified). Let me try again :sweat_smile: 

When the files are provided to `MULTIQC_BY_LANE`, `lane_mqc_files.samples_per_lane` looks like this: `[list of samples for lane 1], [list of samples for lane 2], ...`. The extra multiqc files are needed for each report and need to be included in each of these lists. I thought the best solution would be to use a map over these lists and append the extra files each time, but I could never got that to work.

If you find a better way to perform this operation, I'm all for it :)
